### PR TITLE
Added Manga mode

### PR DESCRIPTION
**Pull request by maydayo**
## PR Checklist

Please check all that apply to this PR using "x":

- [x] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [x] I have checked that this PR does not introduce a breaking change
- [ ] This PR introduces breaking changes and I have provided a detailed explanation below

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting)
- [ ] Refactoring (no functional changes)
- [ ] Documentation changes
- [ ] Other - Please describe:

## Fixes

Issue Number: #10 

## What is the current behavior?

There is no magaMode

## What is the new behavior?

manga mode can be set as options of the reader component
| Name             | Type   | Default | Description                                                      |
| ---------------- | ------ | ------- | ---------------------------------------------------------------- |
| magaMode        | boolean | false    | Set mode to read manga (right to left)                           |

In mangaMode
- the navigation button switch positions
- the slider bar is reverted
- In bookMode the pages change the order to right to left
Screenshot from 2019-10-20 17-23-15

## Other information
Manga Mode state should be displayed to the reader before reading, but I'm not sure where to put it.  Should it put on the first page or somewhere else?

### CODE REVIEW

File: src/components/slider/index.js [hunk 1]
```diff
@@ -82,11 +82,12 @@ class SliderUI extends Component {
     const { max, bufferProgress } = this.props
     const domain = [1, max]
     const { values } = this.state
+    const reversed = this.context.state.mangaMode
```
Comment (btzr-io) [inline src/components/slider/index.js]: this should be a `prop` -> `this.props.reversed`

File: src/components/slider/index.js [hunk 2]
```diff
@@ -116,7 +118,7 @@ class SliderUI extends Component {
             )}
           </Handles>
 
-          <Tracks right={false}>
+          <Tracks right={reversed ? true : false} left={reversed ? false : true}>
```
Comment (btzr-io) [inline src/components/slider/index.js]: this can be more simple:
```JSX
<Tracks right={reversed} left={!reversed}>
```

File: src/index.js [hunk 3]
```diff
@@ -54,6 +54,7 @@ class Villain extends Component {
                     focus={state.focus}
                     container={container}
                     bookMode={state.bookMode}
+                    mangaMode={opts.mangaMode}
```
Comment (btzr-io) [inline src/index.js]: this should probably live  on the context state:
```
state.mangaMode
````

File: src/components/toolbar/navigation.js [hunk 4]
```diff
@@ -38,6 +38,50 @@ class Navigation extends Component {
     }
   }
 
+  renderNavButton = () => {
```
Comment (btzr-io) [inline src/components/toolbar/navigation.js]: We can simpify this function by defining the props as objects:
```JSX
prevButtonProps = {
  title: mangaMode `...` ? 'inverted',
  // Inverted action
  action: mangaMode ? next() : prev()
}

<Button {...prevButtonProps} />
```

**Comment by btzr-io** [comment]
We need a UI component to toggle the manga mode, but that should be a different issue so dont worry about it. Also we are ruining out of space on the toolbar :stuck_out_tongue:

**Comment by maydayo** [comment]
Thanks for reviewing. 

I did the toggle at first, but then I realized that mangaMode has to be set by the uploader. I don't think the reader should be able to control this state. It depends on the type of that manga/pictures. If the comic is not manga (left to right) but mangaMode is turned on, It will confuse the reader. And we cannot flip the image because the letters will be flipped too.

Please let me know what do you think about this?

**Comment by btzr-io** [comment]
@maydayo Check how we hanlde the`autoHideControls` option,
We use the prop to set the default value and the context state to control it.

**Comment by btzr-io** [comment]
Not sure if thats the best way to handle it but, that's what we using for now.
If you have a better solution please share it :v:

**Comment by btzr-io** [comment]
Also could you update change log with your changes?

**Pull request merged**